packages/savings-vault/src/lib/savings-vault.ts

```diff
@@ -39,6 +41,40 @@ export class SavingsVault extends Erc4626<SavingsVaultContractInterface> {
 			: savingsVaultViews;
 	}
 
+	public async redeem(
+		shares: BigNumberish,
+		receiver: Address,
+		owner: Address
+	): Promise<ContractTransaction> {
+		const estimatedGas = await this.contract.estimateGas.redeem(
+			shares,
+			owner,
+			owner
+		);
+
+		return this.contract.redeem(shares, owner, owner, {
+			gasLimit: estimatedGas.mul(100).div(95),
+		});
+	}
+
+	public async redeemWithMaxLoss(
+		shares: BigNumberish,
+		receiver: Address,
+		owner: Address,
+		maxLoss: BigNumber
+	): Promise<ContractTransaction> {
+		const estimatedGas = await this.contract.estimateGas.redeemWithMaxLoss(
+			shares,
+			owner,
+			owner,
+			maxLoss
+		);
+
+		return this.contract.redeemWithMaxLoss(shares, owner, owner, maxLoss, {
```

Comment:
Revert if the passed `maxLoss` is not within `[0-10_000]` range.

Reply:
that's expected, FE will catch that by themselves

Reply:
I know that it will be caught. It is a matter of sending meaningful Errors to clients, like this they will have a hard time debugging if something goes wrong.